lib/package.gi

```diff
@@ -1706,7 +1717,20 @@ InstallGlobalFunction( LoadPackage, function( arg )
 
     od;
 
-    if not IsBound( GAPInfo.LibraryLoaded ) then
+    if IsBound( GAPInfo.LibraryLoaded ) then
+      # Load those package extensions whose condition is satisfied.
+      for i in [ 1 .. Length( GAPInfo.PackageExtensionsPending ) ] do
+        entry:= GAPInfo.PackageExtensionsPending[i];
+        if IsPackageLoaded( entry[1], entry[2] ) then
```

Comment:
What if an extension needs more than one package? Perhaps it should be possible to provide a list of dependencies, not just a single one?